requirements.txt

```diff
@@ -8,6 +8,6 @@
 numpy>=1.16
 Cython>=0.25
 numba
-mpi4py
+mpi4py==3.0.3
```

Comment:
Indeed, this modification may be necessary. According to [this](https://github.com/h5py/h5py/blob/adf2f1104869dcaa6cbc83ecff58229d0c017a6e/setup.py#L38-L49), `h5py` uses the oldest compatible version of `mpi4py` (3.0.3) to build a wheel (which is what we do because we need a parallel build).

Nevertheless, they also say that once the `h5py` wheel is built, it should work with newer versions of `mpi4py`. So maybe we can upgrade `mpi4py` after installing `h5py`...

Do you think that Python 3.10 works after this change?

cc: @anlavandier

Reply:
this change, didn't fix the problem with Python3.10, so we can delete it

Reply:
Please roll back this change, then

Reply:
done